### PR TITLE
Create Github Pull Request  Template

### PR DESCRIPTION
**Pull request by daumie**
For PR  ### What does this PR do?

Creates a Template to be filled when creating Pull Requests

#### Description of Task to be completed?

Add `PULL_REQUEST_TEMPLATE.md` with desired template sections.

#### How should this be manually tested?
After this particular PR has been merged. You should be able to see a Pull Request template for subsequent PR's.

#### Any background context you want to provide?

- Pull request templates help guide developers make better software, and a better software development experience. 
- They are obviously extremely useful for open source projects that get many contributions from outsiders, but they can also be very useful for internal projects where all contributors are insiders. 
- They codify and encourage good behavior, thoughtful code, and empathetic interactions with teammates. 
- They normalize and reinforce a workflow regardless of size, scope, or underlying technology used in a project. 
- And they provide consistency and familiarity that makes switching between multiple projects per day a little easier.

#### What are the relevant Github Issues ?

#48 

#### Screenshots (If applicable)
Here's a peek of the created template

image

### CODE REVIEW

**Review by aonomike — APPROVED** [review]
Thanks again @daumie for this one.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+1. On terminal, run command ...
+
+
+
+#### Any background context you want to provide?
+
+
+
+
+#### What are the relevant Github Issues ?
```
Comment (yakryder) [inline .github/PULL_REQUEST_TEMPLATE.md:23]: Extra whitespace

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+### What does this PR do?
+
+
+
+
+
+#### Description of Task to be completed?
```
Comment (yakryder) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Difference between this and above heading a little hard to tease out

**Comment by yakryder** [comment]
Great change!

**Comment by yakryder** [comment]
@daumie I believe @FedericoEsparza wants us not to delete branches, and I believe the reasoning is that this makes for easier history visualization. This should be doc'ed in `CONTRIBUTING` or equivalent if it isn't

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (daumie) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Thanks for pointing that. I was of the idea that "Description" would be a general a series of steps taken in fulfillment of the Pull Request.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (daumie) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: Thanks, well noted. will make change ASAP.

**Comment by daumie** [comment]
@SherSpock Thanks you for bringing this up.
@aonomike Where can we have this specified?

**Pull request merged**